rllib/policy/tf_policy.py

```diff
@@ -139,6 +143,19 @@ def __init__(self,
             raise ValueError(
                 "seq_lens tensor must be given if state inputs are defined")
 
+        # Apply the post-forward-pass exploration if applicable.
+        # And store the `get_state` op.
+        self._exploration_action = None
+        self._exploration_info = None
+        if self.exploration:
+            self._exploration_action = self.exploration.get_exploration_action(
+                self._action,
+                self.model,
+                action_dist=self.dist_class,
+                explore=self._is_exploring,
+                time_step=self._time_step)
+            self._exploration_info = self.exploration.get_info()
```

Comment:
What's the exploration info for? Is this something we can get on-demand instead of always storing it as an extra variable in the policy?

Reply:
It's for reporting only. Yeah, sure, you could get it on demand (which would be cleaner, code-wise), but then it's an extra session call for the tf-case.

Reply:
The reason I prefer to not store is that it could get out of sync and be a source of bugs. Also, fetching it would be (even more) lightweight if you remove the timestep state from the EpsilonGreedy.

Reply:
Ok. If performance due to smaller session calls is not a concern, I'm happy to change it. It is indeed the cleaner way of doing this.